### ISSUE TITLE
User chooses Map Export Image Size

### ISSUE THREAD

**Issue by AleksueiR** [fgpv-vpgf/fgpv-vpgf#1258]
Give the user an option to pick the image size:
- current map size
- 720p (to be shown as "{$width} x 720" where `width` is calculated using viewer's width/height ratio
- 1080p
- custom height entered by the user (
- [optional/for later] let the config author specify custom presets in the config

Do not scale legend text.

There are two main options of doing this:
- asking user to select the size upfront and then start a print task; 
- defaulting to some size and starting a print task right away; if the user changes the size, need to restart the print job;

**Comment by mweech** [comment]
Can we do a "current size" and then arbitrary presets?

i.e. list would be:

- Current Map Size
- Large (# x #) 
- Medium (# x #) 
- Small  (# x #)

**Comment by AleksueiR** [comment]
I'm almost done with this issue.

It took unexpectedly long because I ended up refactoring most of the export module - it was structured in a very linear way, making it really hard to regenerate all the export component graphics when the map output size is changed. I started working on this before north arrow and scalebar were added to the export image, and needed to incorporate them after the fact (I reused most of the generation code, but in retrospect I should have waited). The export map image module is now structured to easily accept new export components and to let other code modify the behaviour of existing ones (this can be exposed through the public API for the host page to modify the export generation output as it wishes).

The great increase in the number of UI controls (size selector, custom size entry fields, settings toggle, and a list of export components available to be included into the image) needed for the export functionality forced me to redo most of the ui as well. The controls wouldn't fit in to the dialogs action bar and were next to unusable in the mobile layout anyway, so this refactor was going to happen soon anyway. I also had to modify geoApi to actually print map images of different sizes.

Some minor polishing is left and documentation. I plan to wrap everything on Monday to be ready for testing before 1.4 release.
